src/app/modules/bounce/bounce.model.ts

```diff
@@ -132,12 +132,25 @@ BounceSchema.methods.merge = function (
   this.bounces = latestBounces.bounces
 }
 
+/**
+ * Returns true if the document indicates a critical bounce (all recipients
+ * bounced), false otherwise.
+ */
 BounceSchema.methods.isCriticalBounce = function (
   this: IBounceSchema,
 ): boolean {
   return this.bounces.every((emailInfo) => emailInfo.hasBounced)
 }
 
+/**
+ * Returns the list of email recipients for this form
+ */
+BounceSchema.methods.getEmails = function (this: IBounceSchema): string[] {
+  // Return a regular array to prevent unexpected bugs with mongoose
+  // CoreDocumentArray
+  return Array.from(this.bounces.map((emailInfo) => emailInfo.email))
```

Comment:
I think you can simply call [toObject](https://mongoosejs.com/docs/api/array.html#mongoosearray_MongooseArray-toObject) here.

Reply:
Yup I tried that, but `toObject` returns an `any` type, which would have necessitated something like `return (this.toObject() as IBounce).map((emailInfo) => emailInfo.email)`. I felt `Array.from` was cleaner and safer.